common/src/main/java/com/microsoft/identity/common/globalsettings/GlobalSettings.java

```diff
@@ -0,0 +1,256 @@
+        synchronized (mGlobalSettingsSingleton.mGlobalSettingsLock) {
+            if (mGlobalSettingsSingleton.mGlobalSettingsInitialized && !mGlobalSettingsSingleton.mIsDefaulted) {
+                listener.onError(new ClientException(GLOBAL_ALREADY_INITIALIZED_ERROR_CODE,
+                        GLOBAL_ALREADY_INITIALIZED_ERROR_MESSAGE));
+                return;
+            }
+
+            if (mGlobalSettingsSingleton.mIsDefaulted) {
+                listener.onError(new ClientException(GLOBAL_INIT_AFTER_PCA_ERROR_CODE,
+                        GLOBAL_INIT_AFTER_PCA_ERROR_MESSAGE));
+                return;
+            }
+
+            setGlobalConfiguration(
+                    initializeGlobalConfiguration(context, configFileResourceId),
+                    listener
+            );
+        }
+    }
+
+    /**
+     * Load the global configuration file using the configuration file and a listener.
+     *
+     * @param configFile Configuration file.
+     * @param listener Handles success and error messages.
+     */
+    @WorkerThread
+    public static void loadGlobalConfigurationFile(@NonNull final Context context,
+                                                   @NonNull final File configFile,
+                                                   @NonNull final GlobalSettingsListener listener) {
+        synchronized (mGlobalSettingsSingleton.mGlobalSettingsLock) {
+            if (mGlobalSettingsSingleton.mGlobalSettingsInitialized && !mGlobalSettingsSingleton.mIsDefaulted) {
+                listener.onError(new ClientException(GLOBAL_ALREADY_INITIALIZED_ERROR_CODE,
+                        GLOBAL_ALREADY_INITIALIZED_ERROR_MESSAGE));
+                return;
+            }
+
+            if (mGlobalSettingsSingleton.mIsDefaulted) {
+                listener.onError(new ClientException(GLOBAL_INIT_AFTER_PCA_ERROR_CODE,
+                        GLOBAL_INIT_AFTER_PCA_ERROR_MESSAGE));
+                return;
+            }
+
+            setGlobalConfiguration(
+                    initializeGlobalConfiguration(context, configFile),
+                    listener
+            );
+        }
+    }
+
+    /**
+     * Load global configuration file using the default configuration file.
+     * Keeping this private since developers would never need to call this in app, as it's automatically
+     * called in PublicClientApplicationConfiguration creation if developer does not initialize global settings beforehand.
+     *
+     * @param context Context of the App.
+     */
+    @WorkerThread
+    private static void loadDefaultGlobalConfiguration(@NonNull final Context context) {
+        setDefaultGlobalConfiguration(
+                initializeGlobalConfiguration(context)
+        );
+    }
+
+    @WorkerThread
+    private static void setGlobalConfiguration(@NonNull final GlobalSettingsConfiguration globalConfiguration,
+                                               @NonNull final GlobalSettingsListener listener) {
```

Comment:
Just to understand, why don't we return the result as return value? do we expect this method (and other similar methods in this class) to ever be async and be passed result in listener?